src/test_deque.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+from __future__ import unicode_literals
+import pytest
+
+from deque import Deque
+
+APPEND_TABLE1 = [
+    (7),
+    ([1, 2, 3]),
+    ('abc'),
+    (1, 1),
+    ('¥£')
+]
+
+APPEND_TABLE2 = [
+    (None, 1, 1),
+    (7, 4, 2),
+    ('abc', 5, 4),
+    ([1, 2, 3], 5, 4),
+    ((1, 2, 3, 4, 5), [1, 2, 3], 6)
+]
+
+POP_TABLE1 = [
+    (1, 0),
+    ([1, 2, 3], 2),
+    ('abc', 2),
+]
+
+POP_TABLE2 = [
+    ([1, 2, 6, 7], 2),
+    ('abcd', 'b'),
+]
+
+POP_TABLE3 = [
+    (1, 2, 0),
+    ([1, 2, 3], 6, 2),
+    ('abc', 6, 2),
+]
+
+POP_TABLE4 = [
+    (1, 1),
+    ([1, 2, 3], 1),
+    ('abc', 'a'),
+]
+
+POP_TABLE5 = [
+    ([1, 2, 6, 7], 6),
+    ('abcd', 'c'),
+]
+
+POP_TABLE6 = [
+    (1, 1),
+    ([1, 2, 3], 3),
+    ('abc', 'c'),
+]
+
+INIT_TABLE1 = [
+    (None, 0),
+    (7, 1),
+    ('abc', 3),
+    ([1, 2, 3], 3),
+    ((1, 2, 3, 4, 5), 5),
+    ('¥£', 2)
+]
+
+INIT_TABLE2 = [
+    (123, 123),
+    (1, 1),
+    ([1, 2, 3], 3),
+    ('abc', 'c'),
+    ((1, 2, 3, 4, 5), 5),
+    (0, 0),
+    ('¥£', '£')
+]
+
+INIT_TABLE3 = [
+    (123, 123),
+    (1, 1),
+    ([1, 2, 3], 1),
+    ('abc', 'a'),
+    ((1, 2, 3, 4, 5), 1),
+    (0, 0),
+    ('¥£', '¥')
+]
+
+PEEK_TABLE = [
+    ('1', '1'),
+    ('string', 's'),
+    ('$%^', '$'),
+    ([1, 2, 3], 1),
+]
+
+PEEKLEFT_TABLE = [
+    ('1', '1'),
+    ('string', 'g'),
+    ('$%^', '^'),
+    ([1, 2, 3], 3),
+]
+
+
+# Tests for __init__()
+
+
+def test_init_head_None():
+    """Test whether the head_node is None when initiated an empty Deque."""
+    deq = Deque()
+    assert not deq._doublyLL.head_node
+
+
+def test_init_tail_None():
+    """Test whether the tail_node is None when initiated an empty Deque."""
+    deq = Deque()
+    assert not deq._doublyLL.tail_node
```

Comment:
You can be even more explicit by saying `assert deq._doublyLL.tail_node is None`